app/models/manageiq/providers/redfish/physical_infra_manager/operations/power.rb

```diff
@@ -0,0 +1,65 @@
+module ManageIQ::Providers::Redfish
+  module PhysicalInfraManager::Operations::Power
+    # Keep this in sync with app/models/physical_server/operations/power.rb in
+    # core and ResetType enum in Redfish Resource type. Name of the method
+    # comes from the core and the action name used in the reset call from the
+    # ResetType enum.
+    #
+    # NOTE: Not all reset operations are implemented on all servers, so any of
+    # the methods listed here can fail. We need to find a way to let those
+    # failures bubble up to the user interface somehow or risk having a
+    # completely useless tool.
+
+    def power_on(server, options = {})
+      reset_server("On", server, options)
+    end
+
+    def power_off(server, options = {})
+      reset_server("GracefulShutdown", server, options)
+    end
+
+    def power_off_now(server, options = {})
+      reset_server("ForceOff", server, options)
+    end
+
+    def restart(server, options = {})
+      reset_server("GracefulRestart", server, options)
+    end
+
+    def restart_now(server, options = {})
+      reset_server("ForceRestart", server, options)
+    end
+
+    def restart_to_sys_setup(_args, _options = {})
+      $redfish_log.error("Restarting to system setup is not supported.")
```

Comment:
I just noticed the note above the code. Does that note imply that raised errors are not interpreted properly in the UI? I wonder if the notification system should be used? Like https://github.com/ManageIQ/manageiq/pull/15624/files#diff-0811ad92c7ad82e70f192721e7f910d0R56

@gtanzillo what do you think?

Reply:
I raised an exception in my initial implementation here, but since this only fills evm log with useless stack traces with no UI change whatsoever, I changed this to simply log an error message at this stage.

I did implement a quick prototype that uses notification system to inform user about the errors, but in order to have it fully functional, I would need to update core a bit by adding our messages.

Reply:
Thanks, @tadeboro.

I have checked Lenovo provider and they are also not raising any errors. As you pointed out, this should be handled in the core. The PR that I referenced was also used in the end instead of having the Notifications in the provider, so I would keep this as it is for the time being.